core/Pos/Arbitrary/Crypto.hs

```diff
@@ -62,61 +49,33 @@ instance Arbitrary SignTag where
 -- Arbitrary signing keys
 ----------------------------------------------------------------------------
 
--- If you want an arbitrary keypair, just generate a secret key with
--- 'arbitrary' and then use 'Pos.Crypto.toPublic' to get the corresponding
--- public key.
-
-keys :: [(PublicKey, SecretKey)]
-keys = unsafeMakePool "[generating keys for tests...]" 50 keyGen
-{-# NOINLINE keys #-}
-
 instance Arbitrary PublicKey where
-    arbitrary = fst <$> elements keys
+    arbitrary = fst . deterministicKeyGen <$> arbitrarySizedS 32
```

Comment:
I'm afraid that there was a reason for using Secure Random™ here. Though if @neongreen approved, it must be alright.